src/main/webapp/index-script.js

```diff
@@ -499,23 +499,57 @@ function generateReview(review, currUser) {
   upvoteButton.innerHTML += '&#128077;' + review.positive;
   upvoteButton.id += "up" + review.id;
   upvoteButton.addEventListener("click", () => {
-    // Users must be logged in and can only vote once. 
-    if (currUser != null && !review.voters.includes(currUser)) {
-      review.positive += 1;
-      voteOnReview(review);
-      review.voters.push(currUser);
+    // Users must be logged in to vote! 
+    if (currUser != null){
+      // Check if they already voted positive
+      if (!review.positiveVoters.includes(currUser)) {
+        // If not, add their vote. 
+        // Check if they already voted negative
+        if (review.negativeVoters.includes(currUser)){
+          // If so, remove their negative vote.   
+          review.negative -=1;     
+          const index = review.negativeVoters.indexOf(currUser);
+          review.negativeVoters.splice(index, 1);
+        }
+        review.positive += 1;
+        voteOnReview(review);
+        review.positiveVoters.push(currUser);
+      } else {
+        // Otherwise, remove their vote
+        review.positive -= 1;
+        voteOnReview(review);
+        const index = review.positiveVoters.indexOf(currUser);
+        review.positiveVoters.splice(index, 1);
+      }
     }
   });
 
   const downvoteButton = document.createElement('button');
   downvoteButton.innerHTML += '&#128078;' + review.negative;
   downvoteButton.id += "down" + review.id;
   downvoteButton.addEventListener("click", () => {
-    // Users must be logged in and can only vote once. 
-    if(currUser != null && !review.voters.includes(currUser)) {
-      review.negative += 1;
-      voteOnReview(review);
-      review.voters.push(currUser);
+    // Users must be logged in to vote! 
+    if (currUser != null){
+      // Check if they already voted negative
+      if (!review.negativeVoters.includes(currUser)) {
+        // If not, add their vote. 
+        // Check to see if they already voted positive
+        if (review.positiveVoters.includes(currUser)){
+          // If so, remove their vote. 
+          review.positive -=1;
+          const index = review.positiveVoters.indexOf(currUser);
+          review.positiveVoters.splice(index, 1);
+        }
+        review.negative += 1;
+        voteOnReview(review);
```

Comment:
since you call voteOnReview() in both paths, can you just all it outside of the if-else statement?

Reply:
yes

Reply:
great idea - thanks for pointing that out!